crates/cxx-qt-lib/src/core/qflags/mod.rs

```diff
@@ -0,0 +1,342 @@
+/// QFlags is used throughout Qt for storing combinations of enum values.
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[repr(transparent)]
+pub struct QFlags<T: QFlag> {
+    repr: QFlagInt<T>,
+}
+
+impl<T: QFlag> Copy for QFlags<T> {}
+
+impl<T: QFlag> Clone for QFlags<T> {
+    fn clone(&self) -> Self {
+        *self
+    }
+}
+
+impl<T: QFlag> From<T> for QFlags<T> {
+    /// Returns the value stored in the QFlags object as an integer.
+    fn from(value: T) -> Self {
+        Self {
+            repr: value.to_int(),
+        }
+    }
+}
+
+impl<T: QFlag> Default for QFlags<T> {
+    /// Constructs an empty QFlags object.
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl<T: QFlag> QFlags<T> {
+    /// Constructs an empty QFlags object.
+    pub const fn new() -> Self {
+        Self::from_int(T::Repr::ZERO)
+    }
+
+    /// Constructs a QFlags object representing the integer value *i*.
+    pub const fn from_int(i: QFlagInt<T>) -> Self {
+        Self { repr: i }
+    }
+
+    /// Returns the value stored in the QFlags object as an integer.
+    pub const fn to_int(self) -> QFlagInt<T> {
+        self.repr
+    }
+
+    /// Returns `true` if no flag is set (i.e., if the value stored by the QFlags object is 0);
+    /// otherwise returns `false`.
+    pub fn is_empty(self) -> bool {
+        self.repr == T::Repr::ZERO
+    }
+
+    /// Sets the flag *flag* if *on* is `true` or unsets it if *on* is `false`.
+    /// Returns a mutable reference to this object.
+    pub fn set_flag(&mut self, flag: T, on: bool) -> &mut Self {
+        if on {
+            self.repr |= flag.to_int();
+        } else {
+            self.repr &= !flag.to_int();
+        }
+        self
+    }
+
+    /// Returns `true` if the flag `flag` is set, otherwise false.
+    ///
+    /// Note: if *flag* contains multiple bits set to 1 (for instance, if it's an enumerator equal
+    /// to the bitwise-OR of other enumerators) then this function will return `true` if and only if
+    /// all the bits are set in this flags object. On the other hand, if *flag* contains no bits set
+    /// to 1 (that is, its value as a integer is 0), then this function will return `true` if and
+    /// only if this flags object also has no bits set to 1.
+    pub fn test_flag(self, flag: T) -> bool {
+        self.test_flags(Self::from(flag))
+    }
+
+    /// Returns `true` if this *flags* object matches the given flags.
+    ///
+    /// If *flags* has any flags set, this flags object matches precisely if all flags set in
+    /// *flags* are also set in this flags object. Otherwise, when *flags* has no flags set, this
+    /// flags object only matches if it also has no flags set.
+    pub fn test_flags(self, flags: Self) -> bool {
+        if flags.is_empty() {
+            self.is_empty()
+        } else {
+            self.repr & flags.repr == flags.repr
+        }
+    }
+
+    #[cfg(cxxqt_qt_version_at_least_6_2)]
```

Comment:
You can remove the `cfg` flag here and on test_any_flags.

While Qt only supports these functions after 6.2, we can support them for any Qt version, as the logic is still valid and we don't need to call any Qt functions that didn't exist at the time.